RSDKv5/RSDK/Scene/Legacy/v3/CollisionLegacyv3.cpp

```diff
@@ -2152,78 +2152,97 @@ void RSDK::Legacy::v3::ObjectEntityGrip(int32 direction, int32 extendBottomCol,
     collisionRight  = playerHitbox->right[0];
     collisionBottom = playerHitbox->bottom[0];
 
-    int32 p = 0;
-    int32 count = 0;
+    int32 storePos   = 0;
+    int32 storeCount = 0;
 
     scriptEng.checkResult = false;
-    if (collisionStorage[0].entityNo != -1) {
-        count++;
-        scriptEng.checkResult = true;
-    }
-    if (collisionStorage[1].entityNo != -1) {
-        p = 1;
-        count++;
-        scriptEng.checkResult = true;
+
+    for (int32 i = 0; i < COLSTORE_COUNT; i++) {
+        if (collisionStorage[i].entityNo != -1) {
+            storeCount++;
+            storePos              = i;
+            scriptEng.checkResult = true;
+        }
     }
+
     switch (effect) {
         case 0:
         case 1: {
-            if (count) {
+            if (storeCount == 1) {
                 scriptEng.checkResult = false;
                 int32 extendedBottomCol = (extendBottomCol != 0) ? collisionBottom + 5 : -5;
-                Entity *entity          = &objectEntityList[collisionStorage[p].entityNo];
-                int32 yCheck1           = ((collisionStorage[p].top + extendedBottomCol) << 16) + entity->YPos;
-                int32 yCheck2           = ((collisionStorage[p].bottom + extendedBottomCol) << 16) + entity->YPos;
-                int32 xCheck1           = (collisionStorage[p].left << 16) + entity->XPos;
-                int32 xCheck2           = (collisionStorage[p].right << 16) + entity->XPos;
+                Entity *entity          = &objectEntityList[collisionStorage[storePos].entityNo];
+                int32 yCheck1           = ((collisionStorage[storePos].top + extendedBottomCol) << 16) + entity->YPos;
+                int32 yCheck2           = ((collisionStorage[storePos].bottom + extendedBottomCol) << 16) + entity->YPos;
+                int32 xCheck1           = (collisionStorage[storePos].left << 16) + entity->XPos;
+                int32 xCheck2           = (collisionStorage[storePos].right << 16) + entity->XPos;
 
                 if (direction) {
-                    if ((((collisionLeft << 16) + player->XPos) <= xCheck2) && xCheck2 < (player->XPos - player->XVelocity)) {
```

Comment:
I think these parenthesis for clarification were fine: it's a bit harder to read now.